FWCore/Utilities/interface/ESIndices.h

```diff
@@ -32,17 +32,20 @@ namespace edm {
     using Value_t = int;
     constexpr ESResolverIndex() noexcept = default;
     constexpr explicit ESResolverIndex(Value_t iValue) noexcept : index_{iValue} {}
-    constexpr ESResolverIndex(ESResolverIndex const&) noexcept = default;
-    constexpr ESResolverIndex(ESResolverIndex&&) noexcept = default;
-
-    constexpr ESResolverIndex& operator=(ESResolverIndex const&) noexcept = default;
-    constexpr ESResolverIndex& operator=(ESResolverIndex&&) noexcept = default;
```

Comment:
@Dr15Jones Do you know if the compiler-generated copy/move constructors/assignment operators are `constexpr ... noexcept` by default?

Reply:
I'm thinking about this. I could just leave these 4 lines of code the way they were. It is kind of off the topic of the rest of the PR anyway. I just saw a rule of 5 violation (no destructor) and thought these were unnecessary. I'm not sure if the constexpr and noexcept are useful or not... Now that I'm looking the other two classes follow the same pattern.

Reply:
Not the simplest reading and I don't have a copy of the C++20 standard. After about an hour of research on google, I conclude the implicit copy/move constructor/assignment will be constexpr and noexcept if the content of the class allows that. For example, if the class had a data member that did not have a constexpr and noexcept copy constructor, then its implicit copy constructor would not be constexpr and noexcept. And there are lots of complex requirements for constexpr and noexcept to be possible.

In the case of ESResolverIndex, the implicit ones will be constexpr and noexcept.

The only advantage to declaring it explicitly is that if you really wanted that but made a mistake implementing your class (or later when you modified it) such that your class copy constructor could not be noexcept or constexpr, then the compiler would give an error on compilation.

@Dr15Jones I'd be happy to hear your opinion on that.

Reply:
Actually, I don't think the compiler will give any error if you declare noexcept with a function that can throw and it only might give an error or warning for a constexpr function calling a non-constexpr function but the standard does not require a diagnostic... So declaring it explicitly doesn't help much.

Reply:
Matti, Chris, and I discussed this verbally. Summary: Implicitly declared functions would be constexpr and noexcept if used because the content of this class allows that, but it is better to explicitly declare this because the compiler will give useful diagnostic warnings in some cases. Soon, I will push a commit that restores those 4 lines and restart the tests.